cmd/cli/docker/docker_run.go

```diff
@@ -139,101 +134,127 @@ func dockerRun(cmd *cobra.Command, cmdArgs []string, opts *DockerRunOptions) err
 	}
+}
 
-	nodeSelectorRequirements, err := parse.NodeSelector(opts.SpecSettings.Selector)
+func CreateJobDocker(image string, parameters []string, opts *DockerRunOptions) (*models.Job, error) {
+	engineSpec, err := models.DockerSpecBuilder(image).
+		WithParameters(parameters...).
+		WithWorkingDirectory(opts.WorkingDirectory).
+		WithEntrypoint(opts.Entrypoint...).
+		WithEnvironmentVariables(opts.SpecSettings.EnvVar...).Build()
 	if err != nil {
 		return nil, err
 	}
 
-	labels, err := parse.Labels(ctx, opts.SpecSettings.Labels)
-	if err != nil {
-		return nil, err
+	// TODO(forrest) [refactor]: this logic is duplicated in wasm_run
+	resultPaths := make([]*models.ResultPath, 0, len(opts.SpecSettings.OutputVolumes))
+	for name, path := range opts.SpecSettings.OutputVolumes {
+		resultPaths = append(resultPaths, &models.ResultPath{
+			Name: name,
+			Path: path,
+		})
 	}
 
-	spec, err := legacy_job.MakeDockerSpec(
-		image, opts.WorkingDirectory, opts.Entrypoint, opts.SpecSettings.EnvVar, parameters,
-		legacy_job.WithResources(
-			opts.ResourceSettings.CPU,
-			opts.ResourceSettings.Memory,
-			opts.ResourceSettings.Disk,
-			opts.ResourceSettings.GPU,
-		),
-		legacy_job.WithNetwork(
-			opts.NetworkingSettings.Network,
-			opts.NetworkingSettings.Domains,
-		),
-		legacy_job.WithTimeout(opts.SpecSettings.Timeout),
-		legacy_job.WithInputs(opts.SpecSettings.Inputs.Values()...),
-		legacy_job.WithOutputs(outputs...),
-		legacy_job.WithAnnotations(labels...),
-		legacy_job.WithNodeSelector(nodeSelectorRequirements),
-		legacy_job.WithDeal(
-			opts.DealSettings.TargetingMode,
-			opts.DealSettings.Concurrency,
-		),
-	)
+	task, err := models.NewTaskBuilder().
+		Name("TODO").
+		Engine(engineSpec).
+		Publisher(opts.SpecSettings.Publisher.Value()).
+		ResourcesConfig(&models.ResourcesConfig{
+			CPU:    opts.ResourceSettings.CPU,
+			Memory: opts.ResourceSettings.Memory,
+			Disk:   opts.ResourceSettings.Disk,
+			GPU:    opts.ResourceSettings.GPU,
+		}).
+		InputSources(opts.SpecSettings.Inputs.Values()...).
+		ResultPaths(resultPaths...).
+		Network(&models.NetworkConfig{
+			Type:    opts.NetworkingSettings.Network,
+			Domains: opts.NetworkingSettings.Domains,
+		}).
+		Timeouts(&models.TimeoutConfig{ExecutionTimeout: opts.SpecSettings.Timeout}).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create job: %w", err)
+	}
 
-	// Publisher is optional and we won't provide it if not specified
-	p := opts.SpecSettings.Publisher.Value()
-	if p != nil {
-		spec.Publisher = p.Type //nolint:staticcheck
-		spec.PublisherSpec = *p
+	labels, err := parse.StringSliceToMap(opts.SpecSettings.Labels)
+	if err != nil {
+		return nil, fmt.Errorf("parseing job labels: %w", err)
 	}
 
+	constraints, err := parse.NodeSelector(opts.SpecSettings.Selector)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("parseing job contstrints: %w", err)
+	}
+	job := &models.Job{
+		Name:        "TODO",
+		Namespace:   "TODO",
```

Comment:
Similar point to task name in my comment above - these could be UUIDs, values from flags we add, or something else?

Reply:
we have a default namespace for `bacalhau job run` https://github.com/bacalhau-project/bacalhau/blob/6f00c8a89e23b28a882e467885d1b6074f81f63e/pkg/models/constants.go#L5

though we are still using clientID for `docker run`. I think we should use the default namespace, but expose a flag to set the desired namespace. This means users won't be able to filter their own jobs in the public network, but that might be acceptable based on our longer term plans. Thoughts?